daras_ai_v2/base.py

```
@@ -2082,9 +2082,11 @@ def estimate_run_duration(self) -> int | None:
         pass
 
     def submit_and_redirect(
-        self, unsaved_state: dict[str, typing.Any] = None
+        self,
+        unsaved_state: dict[str, typing.Any] | None = None,
+        **defaults,
     ) -> typing.NoReturn | None:
```

Ruff (0.15.21)

[warning] `NoReturn | T` is equivalent to `T`

Remove `NoReturn`

(RUF020)